packages/features/settings/OutOfOfficeEntriesListForTeam.tsx

```diff
@@ -0,0 +1,280 @@
+import { keepPreviousData } from "@tanstack/react-query";
+import type { ColumnDef } from "@tanstack/react-table";
+import { Trans } from "next-i18next";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { useFormState } from "react-hook-form";
+
+import dayjs from "@calcom/dayjs";
+import { getUserAvatarUrl } from "@calcom/lib/getAvatarUrl";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Avatar, Button, DataTable, EmptyScreen, Icon, showToast, SkeletonText, Tooltip } from "@calcom/ui";
+import type { BookingRedirectForm } from "@calcom/web/pages/settings/my-account/out-of-office";
+
+interface OutOfOfficeEntry {
+  id: number;
+  uuid: string;
+  start: Date;
+  end: Date;
+  toUserId: number | null;
+  toUser: {
+    username: string;
+  } | null;
+  reason: {
+    id: number;
+    emoji: string;
+    reason: string;
+    userId: number;
+  } | null;
+  notes: string | null;
+  user: { id: number; avatarUrl: string; username: string; email: string } | null;
+}
+
+export const OutOfOfficeEntriesListForTeam = ({
+  editOutOfOfficeEntry,
+  teamOOOEntriesUpdated,
+}: {
+  editOutOfOfficeEntry: (entry: BookingRedirectForm) => void;
+  teamOOOEntriesUpdated: number;
+}) => {
+  const { t } = useLocale();
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const [debouncedSearchTerm, setDebouncedSearchTerm] = useState("");
+  const [deletedEntry, setDeletedEntry] = useState(0);
+  const { data, isPending, fetchNextPage, isFetching, refetch } =
+    trpc.viewer.outOfOfficeEntriesList.useInfiniteQuery(
+      {
+        limit: 10,
+        fetchTeamMembersEntries: true,
+        searchTerm: debouncedSearchTerm,
+      },
+      {
+        getNextPageParam: (lastPage) => lastPage.nextCursor,
+        placeholderData: keepPreviousData,
+      }
+    );
+
+  useEffect(() => {
+    refetch();
+  }, [teamOOOEntriesUpdated, deletedEntry, refetch]);
+
+  const totalDBRowCount = data?.pages?.[0]?.meta?.totalRowCount ?? 0;
+  //Flatten the array of arrays from the useInfiniteQuery hook
+  const flatData = useMemo(
+    () => data?.pages?.flatMap((page) => page.rows) ?? [],
+    [data]
+  ) as OutOfOfficeEntry[];
+  const totalFetched = flatData.length;
+
+  const columns: ColumnDef<OutOfOfficeEntry>[] = [
+    {
+      id: "member",
+      header: `Member`,
+      cell: ({ row }) => {
+        if (!row.original || !row.original.user) {
+          return <SkeletonText className="h-8 w-full" />;
+        }
+        const { avatarUrl, username, email } = row.original.user;
+        return (
+          <div className="flex items-center gap-2">
+            <Avatar
+              size="sm"
+              alt={username || email}
+              imageSrc={getUserAvatarUrl({
+                avatarUrl,
+              })}
+            />
+            <div className="">
+              <div
+                data-testid={`member-${username}-username`}
```

Comment:
This is similar to the `OutOfOfficeEntriesList`. No need to create another component, instead use it accordingly.